invenio_app_ils/items/api.py

```diff
@@ -60,16 +59,23 @@ def ensure_item_can_be_updated(self, record):
         """Raises an exception if the item's status cannot be updated."""
         latest_version = record.revisions[-1]
         if latest_version:
-            status = latest_version.get("status", None)
-        else:
-            status = None
+            latest_version_document_pid = latest_version.get("document_pid", None)
         pid = record["pid"]
-        if status == "CAN_CIRCULATE":
-            item_pid = dict(value=pid, type=ITEM_PID_TYPE)
-            active_loan = get_active_loan_by_item_pid(item_pid).execute().hits
-            total = active_loan.total if lt_es7 else active_loan.total.value
-            if total > 0:
-                raise ItemHasActiveLoanError(active_loan[0]["pid"])
+        item_pid = dict(value=pid, type=ITEM_PID_TYPE)
+        loan_search_res = search_by_pid(item_pid=item_pid)
+        aggs = A("terms", field="state")
```

Comment:
I would avoid aggregations because they can be slow.
Why not instead:
```
if invenio_circulation.search.api.search_by_pid(
   item_pid=item_pid,
   filter_states=current_app.config["CIRCULATION_STATES_LOAN_ACTIVE"]
          + current_app.config["CIRCULATION_STATES_LOAN_COMPLETED"]
          + current_app.config["CIRCULATION_STATES_LOAN_CANCELLED"],
).count():
   raise
```